sys/net/gnrc/link_layer/lwmac/include/lwmac_internal.h

```diff
@@ -0,0 +1,150 @@
+#define GNRC_LWMAC_INTERNAL_H_
+
+#include <stdint.h>
+#include "periph/rtt.h"
+#include "lwmac_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* @brief   Type to pass information about parsing */
+typedef struct {
+    lwmac_hdr_t* header;    /**< lwmac header of packet */
+    l2_addr_t  src_addr;    /**< copied source address of packet  */
+    l2_addr_t  dst_addr;    /**< copied destination address of packet */
+} lwmac_packet_info_t;
+
+/* @brief   Next RTT event must be at least this far in the future
+ *
+ * When setting an RTT alarm to short in the future it could be possible that
+ * the counter already passed the calculated alarm before it could be set. This
+ * margin will be applied when using `_next_inphase_event()`.
+ */
+#define LWMAC_RTT_EVENT_MARGIN_TICKS    ( RTT_MS_TO_TICKS(2) )
+
+/* @brief Extract the destination address out of an GNRC_NETTYPE_NETIF pktsnip
+ *
+ * @param[in]   pkt                 pktsnip from whom to extract
+ * @param[out]  pointer_to_addr     pointer to address will be stored here
+ *
+ * @return                          length of destination address
+ */
+int _get_dest_address(gnrc_pktsnip_t* pkt, uint8_t* pointer_to_addr[]);
+
+/* @brief Find the first pktsnip of @p type
+ *
+ * Will search linearly through the packet buffer @p pkt and yield
+ * gnrc_pktsnip_t::data of the first pktsnip match the type @p type.
+ *
+ * @param[in]   pkt     pktsnip that will be searched
+ * @param[in]   type    type to search for
+ *
+ * @return              pointer to data, NULL is not found
+ */
+void* _gnrc_pktbuf_find(gnrc_pktsnip_t* pkt, gnrc_nettype_t type);
+
+/* @brief Parse an incoming packet and extract important information
+ *
+ * Copies addresses into @p info, but header points inside @p pkt.
+ *
+ * @param[in]   pkt             packet that will be parsed
+ * @param[out]  info            structure that will hold parsed information
+ *
+ * @return                      0 if correctly parsed
+ * @return                      <0 on error
+ */
+int _parse_packet(gnrc_pktsnip_t* pkt, lwmac_packet_info_t* info);
+
+
+/* @brief Shortcut to get the state of netdev
+ *
+ * @param[in]   lwmac           lwmac state that stores netdev pointer
+ *
+ * @return                      state of netdev
+ */
+netopt_state_t _get_netdev_state(lwmac_t* lwmac);
+
+/* @brief Shortcut to set the state of netdev
+ *
+ * @param[in]   lwmac           lwmac state that stores netdev pointer
+ * @param[in]   devstate        new state for netdev
+ */
+void _set_netdev_state(lwmac_t* lwmac, netopt_state_t devstate);
+
+/* @brief Check if packet is broadcast
+ *
+ * @param[in]   pkt             packet to check
+ */
+static inline bool _packet_is_broadcast(gnrc_pktsnip_t* pkt)
+{
+    gnrc_netif_hdr_t* netif_hdr = _gnrc_pktbuf_find(pkt, GNRC_NETTYPE_NETIF);
+    return ( (netif_hdr == NULL) ? false :
+                              (netif_hdr->flags & (GNRC_NETIF_HDR_FLAGS_BROADCAST | GNRC_NETIF_HDR_FLAGS_MULTICAST)) );
```

Comment:
I found that broadcast packet will be regarded as data packet, namely, this function can not return the right value. For example, when I typed ```txtsnd 4 ff:ff Helloworld``` (tested with the netdev_1 version, I have also tried with ```ff:ff:ff:ff:ff:ff:ff:ff``` with the netdev2 version), I found that  Lw_MAC will put this packet into the queue of ```neighbor[i], i>0```, while broadcast packets are intended to be put into the queue of  the first neghbor, i.e., ```neighbor[0]```. Then,  Lw_MAC will first send out a stream of WR packets (for sending this broadcast packet), trying to notice a node of ```ff:ff```. 

So, where does the ```netif_hdr->flags``` be set to have ```GNRC_NETIF_HDR_FLAGS_BROADCAST and GNRC_NETIF_HDR_FLAGS_MULTICAST```? Is it the job of a upper layer when the packet is generated? Or, the related flags should be set by the MAC layer (Lw_MAC)?  (Sorry I am not very familiar about GNRC yet.)